File: 9/main.js
/**
 * @param {number} x
 * @return {boolean}
 */
const isPalendrome = (x) => {
  if (x < 0) return false;
  let string = x.toString();
  if (string.length < 2) return true;
  if (string.length % 2 !== 0) {
    string = string
      .slice(0, Math.floor(string.length / 2)) + string.slice(Math.floor(string.length / 2) + 1);
  }
  while (string.length !== 0) {
    if (string[0] !== string[string.length - 1]) {
      return false;
    }
    string = string.substring(1, string.length - 1);
  }
  return true;
};

// TEST SUITE

// TEST 1

function test1() {
  const actual = isPalendrome(121);
  const expected = true;
  if (actual === expected) {
    return '✔';
  }
  return 'X';
}

console.log(test1());

// TEST 2

function test2() {
  const actual = isPalendrome(-121);
  const expected = false;
  if (actual === expected) {
    return '✔';
  }
  return 'X';
}

console.log(test2());

// TEST 3

function test3() {
  const actual = isPalendrome(10);
  const expected = false;
  if (actual === expected) {
    return '✔';
  }
  return 'X';
}

console.log(test3());

File: 9/main.py
import math

def isPalendrome(x):
  if x < 0:
    return False
  x = str(x)
  if len(x) < 2:
    return True
  if len(x) % 2 != 0:
    x = x[0:math.floor(len(x) / 2)] + x[math.floor(len(x) / 2) + 1:5]
  while len(x) != 0:
    if (x[0] != x[len(x) - 1]):
      return False
    else:
      x = x[1:len(x) - 1]
  return True


# TEST SUITE

# TEST 1

print(isPalendrome(121))

# TEST 2

print(isPalendrome(-121))

# TEST 1

print(isPalendrome(10))

